cmd/jpgo/main.go

```diff
@@ -0,0 +1,92 @@
+/*Basic command line interface for debug and testing purposes.
+
+Examples:
+
+Only print the AST for the expression:
+
+    jp.go -ast "foo.bar.baz"
+
+Evaluate the JMESPath expression against JSON data from a file:
+
+    jp.go -input /tmp/data.json "foo.bar.baz"
+
+This program can also be used as an executable to the jp-compliance
+runner (github.com/jmespath/jmespath.test).
+
+*/
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+import (
+	"encoding/json"
+
+	"github.com/jmespath/jmespath.go/jmespath"
+	"github.com/kr/pretty"
+)
+
+func errMsg(msg string, a ...interface{}) int {
+	fmt.Fprintf(os.Stderr, msg, a...)
+	fmt.Fprintln(os.Stderr)
+	return 1
+}
+
+func run() int {
+
+	astOnly := flag.Bool("ast", false, "Print the AST for the input expression and exit.")
+	inputFile := flag.String("input", "", "Filename containing JSON data to search. If not provided, data is read from stdin.")
+
+	flag.Parse()
+	args := flag.Args()
+	if len(args) != 1 {
+		return errMsg("Expected a single argument (the JMESPath expression).")
```

Comment:
This error probably should also include the flags.PrintDefaults() method which will print out commandline arguments

http://tip.golang.org/pkg/flag/#PrintDefaults